### PR TITLE
Fix tableView error in Unit Tests.

### PR DESCRIPTION
**Pull request by Ivan-Skoryk**
## Summary
Fixed error with table view in Unit Test.

 #### Checklist:
 - [x] I have added a brief description of the problem in the PR body.
 - [x] I have explained how I solved the problem in the PR body.
 - [x] I have covered possible edge cases and side effects.
 - [x] I have have attached screenshots to the PR of any UI change.
 - [x] I have checked that my changes work on iOS and macOS targets.
 - [x] I have checked that my changes work correctly on devices with and without a Home button.
 - [x] I have added unit tests for the new logic.
 - [x] I have achieved maximum possible code coverage for the new code.
 - [x] I have checked that my UI works correctly on both Light and Dark themes.

### CODE REVIEW

File: xDripTests/Scenes/Entries/EntriesList/Logic/EntriesListTableControllerTests.swift [hunk 1]
```diff
@@ -97,26 +85,24 @@ class EntriesListTableControllerTests: XCTestCase {
...
-        sut.reload(with: data)
-        tableView.reloadData()
-        
-        let indexPath = IndexPath(row: 10, section: 0)
-        
-        sut.tableView(tableView, commit: .delete, forRowAt: indexPath)
-        
-        XCTAssertTrue(calledDelete)
-        XCTAssertTrue(tableView.numberOfRows(inSection: 0) == 19)
+        dispatchAfter(seconds: 1) {
```
Comment (Artem-Kalmykov) [inline xDripTests/Scenes/Entries/EntriesList/Logic/EntriesListTableControllerTests.swift:88]: 1. Isn't regular `DispatchQueue.main.async {}` enough? It should dispatch it on the next run loop, which should fix the issue.
2. If you add async operations, you also need to add expectations to prevent tests to end early.

**Review by Artem-Kalmykov — CHANGES_REQUESTED** [review]

File: xDripTests/Scenes/Entries/EntriesList/Logic/EntriesListTableControllerTests.swift (hunk 1 above)
Comment (Ivan-Skoryk) [reply xDripTests/Scenes/Entries/EntriesList/Logic/EntriesListTableControllerTests.swift:88]: I've tried regular `DispatchQueue.main.async {}`, it didn't fix issue.

Comment (Artem-Kalmykov) [reply xDripTests/Scenes/Entries/EntriesList/Logic/EntriesListTableControllerTests.swift:88]: Try a different approach, like adding it on a view controller first, similar to view controller unit tests.
